src/Core/Form/IdentifiableObject/DataFormatter/AbstractFormDataFormatter.php

```diff
@@ -57,4 +68,42 @@ protected function formatByPath(array $formData, array $pathAssociations): array
 
         return $formattedData;
     }
+
+    protected function formatMultiShopAssociation(string $originalField, string $formattedField): array
```

Comment:
@jolelievre could you check at this?
I wonder if its a good decision? Also bellow to method `prefixWithModifyAllShops`, I wonder if I could improve it and also if I should extract it to some "string util" component which could be unit tested.

Reply:
About the string util I guess you mean that the `prefixWithModifyAllShops` is the same logic as in the `CommandBuilder` then indeed it would be interesting to extract this common logic in an util static method for example that can be unit tested. So it should also be used in the command builder then.

Regarding the implementation itself, it will not be the same anymore since yours is based on pure string whereas the initial one was based on PropertyPath that you don't have here. So there are two possible implementations for this class util:
- based on string path
- based on property path

I liked the property path implementation but honestly, it's a detail so I don't mind if you use either of these as long as it's tested and it works none is better than the other. Maybe the string approach is actually more straightforward and reusable.

TLDR; you would need something like `ModifyAllShopsUtil::prefix(string $propertyPath, string $prefix): string`